### PR TITLE
Added tiff support throughout udStream

### PR DESCRIPTION
**Pull request by int-Frank**
Fixes [AB#2281](https://dev.azure.com/euclideon/57c3bd2a-8f94-4578-b195-2e4fa9d8f295/_workitems/edit/2281)

### CODE REVIEW

File: vcGL/src/vcTextureHelper.cpp [hunk 1]
```diff
@@ -414,3 +415,62 @@ udResult vcTexture_SaveImage(vcTexture *pTexture, vcFramebuffer *pFramebuffer, c
...
+    if (pPixelData != nullptr)
+      break;
+
+    pPixelData = vcTexture_DecodePixels_tiff(pFileContents, fileLength, pImageData);
+    if (pPixelData != nullptr)
+      break;
+
+  } while (false);
+
+  return pPixelData;
```
Comment (OsiJr) [inline vcGL/src/vcTextureHelper.cpp:475]:
```suggestion
  uint8_t *pPixelData = nullptr;

  // try stbi
  if (pPixelData == nullptr)
    pPixelData = vcTexture_DecodePixels_stbi(pFileContents, fileLength, pImageData);

  // if nothing loaded, try libtiff
  if (pPixelData == nullptr)
    pPixelData = vcTexture_DecodePixels_tiff(pFileContents, fileLength, pImageData);
    
  return pPixelData;
```

Is a bit cleaner and

File: vcGL/src/vcTiff.cpp [hunk 2]
```diff
@@ -0,0 +1,189 @@
...
+  vcMemoryInStream *pStream = nullptr;
+  TIFF *pTif = nullptr;
+  size_t npixels = 0;
+  uint32 w, h;
+  uint32 *pRaster = nullptr;
+
+  UD_ERROR_NULL(pBuffer, udR_InvalidParameter_);
+  UD_ERROR_NULL(pW, udR_InvalidParameter_);
+  UD_ERROR_NULL(pH, udR_InvalidParameter_);
+  UD_ERROR_NULL(ppPixelData, udR_InvalidParameter_);
```
Comment (OsiJr) [inline vcGL/src/vcTiff.cpp:161]: The convention for `vc*` is to just have this as an if check above everything else.

File: src/scene/vcMedia.cpp [hunk 3]
```diff
@@ -148,9 +148,9 @@ void vcMedia::AddToScene(vcState *pProgramState, vcRenderData *pRenderData)
     udInt2 imageSize = {};
     if (m_pImageData != nullptr)
```
Comment (OsiJr) [inline src/scene/vcMedia.cpp:149]: In general, this file looks like it should have use the texture cache instead (outside the scope of this PR)

Comment (int-Frank) [reply src/scene/vcMedia.cpp:149]: Ok. Is adapting `vcMedia` to use the texture cache within the scope of this PR? As I write this I would imagine it should probably be it's own PR.

Comment (int-Frank) [reply src/scene/vcMedia.cpp:149]: Should I roll this back?

**Review by OsiJr — APPROVED** [review]

**Pull request merged**